CanDIG/CanDIGv2
# Unauthorized user is redirected to 'request access' page

# Jira Ticket
[DIG-1857](https://candig.atlassian.net/browse/DIG-1857)

## Sub PRs:

https://github.com/CanDIG/candigv2-ingest/pull/147
https://github.com/CanDIG/candig-data-portal/pull/201

# Description
This allows a logged-in user to see a request access page if they are not yet in OPA, and to request access.

# Screenshots
![image](https://github.com/user-attachments/assets/d76b8eee-1cf7-4417-95c0-a36fc9c95d4f)

![image](https://github.com/user-attachments/assets/b1bd7484-2a02-4063-94d4-0b2e4be5fe1c)

# To Test
- Delete the site admin from the authorized users list like so:
```
(candig) 16:13 fnguyen@fnguyen-worktop:~/Documents/CanDIGv2$ docker exec -it candigv2_candig-ingest_1 python
Python 3.12.8 (main, Dec  4 2024, 20:39:47) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import authx.auth
>>> authx.auth.delete_service_store_secret("opa", key=f"users/site_admin@test.ca")
204
```
- Login to the data portal, check that you are totally unauthorized
- Check that you are unauthorized via the ingest API like so:
```
(candig) 13:18 fnguyen@fnguyen-worktop:~/Documents/CanDIGv2$ curl candig.docker.internal:5080/ingest/user/pending -H "Authorization: Bearer $TOKEN"                                     
{                                                                                                                                                                                       
  "results": [                                                                                                                                                                          
    "site_admin@test.ca"                                                                                                                                                                
  ]                                                                                                                                                                                     
}
```
- Click the button to request access, the page should change to reflect that you are now awaiting authorization
- Authorize the pending user like so:
```
(candig) 17:17 fnguyen@fnguyen-worktop:~/Documents/CanDIGv2$ curl -X POST candig.docker.internal:5080/ingest/user/pending/site_admin@test.ca -H "Authorization: Bearer $TOKEN"
{
  "pending_users": {}
}
```
- Refresh your page on the frontend and you should be in.

[DIG-1857]: https://candig.atlassian.net/browse/DIG-1857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
